src/libstd/sys_common/mod.rs

```diff
@@ -73,6 +74,7 @@ cfg_if::cfg_if! {
     if #[cfg(any(target_os = "cloudabi",
                  target_os = "l4re",
                  target_os = "hermit",
+                 feature = "restricted-std",
```

Comment:
I sort of forget what's going on here, could you remind me what's up with this module?

This may be a good candidate for movement to `sys` instead of `sys_common`?

Reply:
I'm not sure I fully understand the question.

There are some things defined in `sys_common::net` that are used in various other places throughout std. In particular:

- `libstd/net/addr.rs`: Needs `LookupHost`
- `libstd/net/tcp.rs`: Needs `TcpStream` and `TcpListener`
- `libstd/net/udp.rs`: Needs `UdpSocket`

`sys_common/net.rs` has implementations that work for *almost* every platform, except those listed above.  cloudabi, wasm, and bare-metal targets don't support these (they have stub implementations).  l4re, hermit, and sgx have custom implementations.

If it is moved to `sys`, I think it would need to use `#[path]` tricks to share amongst unix, windows, vxworks, wasi, etc.   I have no objection to that, but I don't fully understand the philosophy of sys_common.  A lot of stuff in there does not appear to be 100% common, so I'm not sure how the line gets drawn.

Reply:
Hm ok so yeah if you're up for it I think the best way to deal with this is to use `#[path]` tricks like you are for other platforms. That'll make it easy to avoid code duplication while also making it easy to allow each platform to choose its own implementation.

Otherwise the original intention of `sys_common` was basically what it says on the tin, everything in common between all `sys` modules. That being said as you've realized this rarely holds up to be true over time. When you add a new platform that needs *almost* everything but not quite, it's way easier to add a few `#[cfg]` than to reorganize to align with the original principles.

Nowadays I sort of feel like `sys_common` should either be 100% (or maybe 99%) shared or otherwise everything should be shared with `#[path]`.

Reply:
Would you mind if I defer that to a separate PR? I'd actually like to do a pass of de-duplicating large sections of the sys module, and I think this would fit with that.  In particular, there are large chunks of duplicate code in the cloudabi, hermit, sgx, and vxworks modules. It is not clear to me if there was a good reason for that, do you happen to know? Do you think there would be any objections to adding more `#[path]` attributes to share amongst those?

(This is a practical thing, I was recently looking at making a small change to `Command`, and saw that I needed to update several identical files.)

Reply:
Ah yeah that's ok, a separate PR should work fine 👍

Reply:
If you could, please cc me on the follow-up PR.